homeassistant/components/alarm_control_panel/arlo.py

```diff
@@ -0,0 +1,117 @@
+"""
+This component provides HA alarm_control_panel support for Arlo System.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/alarm_control_panel.arlo/
+"""
+import asyncio
+import logging
+import voluptuous as vol
+
+from homeassistant.components.arlo import (DATA_ARLO, CONF_ATTRIBUTION)
+from homeassistant.components.alarm_control_panel import (AlarmControlPanel,
+                                                          PLATFORM_SCHEMA)
+from homeassistant.const import (ATTR_ATTRIBUTION, STATE_ALARM_ARMED_AWAY,
+                                 STATE_ALARM_ARMED_HOME, STATE_ALARM_DISARMED)
+import homeassistant.helpers.config_validation as cv
+
+DEPENDENCIES = ['arlo']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_HOME_MODE_NAME = 'home_mode_name'
+ICON = 'mdi:security'
+ARMED = "armed"
+DISARMED = "disarmed"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOME_MODE_NAME, default=ARMED): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up a sensor for an Arlo device."""
+    data = hass.data[DATA_ARLO]
+
+    if not data.base_stations:
+        return
+
+    home_mode_name = config.get(CONF_HOME_MODE_NAME)
+    base_stations = []
+    for base_station in data.base_stations:
+        base_stations.append(ArloBaseStation(base_station, home_mode_name))
+    async_add_devices(base_stations)
+
+
+class ArloBaseStation(AlarmControlPanel):
+    """An alarm_control_panel implementation for Arlo."""
+
+    def __init__(self, data, home_mode_name):
+        """Initialize the alarm control panel."""
+        self._base_station = data
+        self._home_mode_name = home_mode_name
+        self._state = self._base_station.mode
```

Comment:
Init this as `None`, and make sure update is called after adding entities, like this:
```py
async_add_devices(devices, True)
```

Reply:
Done.